worker/src/runner/container_rt.rs

```diff
@@ -31,23 +35,45 @@ impl ContainerRuntime {
             {
                 let error = e.to_string();
                 handle
-                    .report_instance_status(Status::FailedToStart { error })
+                    .report_instance_status(spec.instance_id, Status::FailedToStart { error })
                     .await;
                 return;
             }
 
-            // healthcheck verifies if service is running on established `PORT`
-            handle.report_instance_status(Status::Started).await;
+            // Healthcheck verifies if service is running on established `PORT`
+            handle
+                .report_instance_status(spec.instance_id, Status::Started)
+                .await;
 
             if let Err(e) = this.wait_container(&container_name).await {
                 let error = e.to_string();
                 handle
-                    .report_instance_status(Status::Crashed { error })
+                    .report_instance_status(spec.instance_id, Status::Crashed { error })
                     .await;
                 return;
             }
 
-            handle.report_instance_status(Status::Terminated).await;
+            handle
+                .report_instance_status(spec.instance_id, Status::Terminated)
+                .await;
+        });
+    }
+
+    pub fn terminate_instance(&self, id: InstanceId) {
+        let this = self.clone();
+        tokio::spawn(async move {
+            let name = format!("instance-{id:?}");
```

Comment:
```suggestion
            let name = format!("instance-{id}");
```
You shouldn't use debug formatting for this purpose. In general, aren't stable

Reply:
`InstanceId` doesn't implement `std::fmt::Display`, may I use `{:#?}`

Reply:
I've decided to implement `fmt::Display` for `InstanceId`, came out better tbh!

Reply:
SGTM